### PR TITLE
BPKHotelStarRating UIKit

### PR DESCRIPTION
**Pull request by iHandle**
BPKHotelStarRating in UIKit Version

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `README.md`
+ [x] Tests
+ [x] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [ ] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/StarRating/Classes/BPKHotelStarRating.swift [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        NSLayoutConstraint.activate([
+            stackView.topAnchor.constraint(equalTo: topAnchor),
+            stackView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            stackView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            stackView.trailingAnchor.constraint(equalTo: trailingAnchor)
+        ])
+    }
+    
+    private func updateLookAndFeel() {
+        stackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
```
Comment (gert-janvercauteren) [inline Backpack/StarRating/Classes/BPKHotelStarRating.swift]: We also need to call stackView.removeArrangedSubview() 
Learn more: https://stackoverflow.com/a/53260606

File: Backpack/StarRating/Classes/BPKHotelStarRating.swift [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        NSLayoutConstraint.activate([
+            stackView.topAnchor.constraint(equalTo: topAnchor),
+            stackView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            stackView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            stackView.trailingAnchor.constraint(equalTo: trailingAnchor)
+        ])
+    }
+    
+    private func updateLookAndFeel() {
+        stackView.arrangedSubviews.forEach({ $0.removeFromSuperview() })
```
[the remark this answers is not in the document]
Comment (iHandle) [reply Backpack/StarRating/Classes/BPKHotelStarRating.swift]: Updated at [8dc5127](https://github.com/Skyscanner/backpack-ios/pull/1674/commits/8dc51275095bd3bc6a8a959448463e307d9a6f43)

File: Backpack/StarRating/Classes/BPKHotelStarRating.swift [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import UIKit
+
+public class BPKHotelStarRating: UIView {
```
Comment (gert-janvercauteren) [inline Backpack/StarRating/Classes/BPKHotelStarRating.swift:23]: Would it make sense to make this a UIStackView subview? @frugoman 
That way we remove a layer of nesting?

Comment (iHandle) [reply Backpack/StarRating/Classes/BPKHotelStarRating.swift:23]: It make sense. That also makes the component simpler :)

Comment (frugoman) [reply Backpack/StarRating/Classes/BPKHotelStarRating.swift]: Agree on this one! good catch!

File: Backpack/StarRating/Classes/BPKHotelStarRating.swift [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    private func setup() {
+        translatesAutoresizingMaskIntoConstraints = false
+        axis = .horizontal
+        updateLookAndFeel()
+    }
+    
+    private func updateLookAndFeel() {
+        arrangedSubviews.forEach { view in
+            removeArrangedSubview(view)
+            view.removeFromSuperview()
```
Comment (frugoman) [inline Backpack/StarRating/Classes/BPKHotelStarRating.swift:69]: Do you need to call `removeFromSuperview()` after removing from the stackView? I think you dont need to do it, but please check!

if so: code could look like this
```
arrangedSubviews.forEach(removeArrangedSubview) // passing the method reference!
```swift

**Review by frugoman — APPROVED** [review]

File: Backpack/StarRating/Classes/BPKHotelStarRating.swift (hunk 4 above)
Comment (frugoman) [reply Backpack/StarRating/Classes/BPKHotelStarRating.swift:69]: This is needed, nevermind!

Comment (iHandle) [reply Backpack/StarRating/Classes/BPKHotelStarRating.swift:69]: As @gert-janvercauteren  mentioned earlier, we do need to call the removeArrangedSubview
Stackoverflow: https://stackoverflow.com/a/53260606
Apple's document: https://developer.apple.com/documentation/uikit/uistackview/1616235-removearrangedsubview

**Pull request merged**